src/main/java/org/folio/services/validator/InvoiceValidator.java

```diff
@@ -119,42 +126,76 @@ private void validateInvoiceLineFundDistributions(List<InvoiceLine> invoiceLines
-        throw new HttpException(400, LINE_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
-      }
+      validateFundDistributionForInvoiceLine(line.getSubTotal(),line.getFundDistributions(),currencyUnit);
     }
   }
 
-  private boolean isFundDistributionSummaryNotValid(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return ObjectUtils.notEqual(sumMixedDistributions(line, currencyUnit), line.getTotal());
-  }
 
-  private Double sumMixedDistributions(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return sumMixedDistributions(line.getFundDistributions(), line.getTotal(), currencyUnit);
-  }
-
-  private void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments, CurrencyUnit currencyUnit) {
-    for (Adjustment adjustment : adjustments){
+  public static void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments,CurrencyUnit currencyUnit) {
+    for (Adjustment adjustment : adjustments) {
       if (CollectionUtils.isEmpty(adjustment.getFundDistributions())) {
         throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_NOT_PRESENT);
       }
-
-      if (isFundDistributionSummaryNotValid(adjustment, currencyUnit)) {
-        throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
+      validateFundDistributionForInvoiceLine(adjustment.getValue(), adjustment.getFundDistributions(), currencyUnit);
+    }
+  }
+  public static void validateFundDistributionForInvoiceLine(Double total, List<FundDistribution> fundDistributions,CurrencyUnit currencyUnit) {
+    Double subtotal = Money.of(total,currencyUnit).getNumber().doubleValue();
+    if (subtotal != null && CollectionUtils.isNotEmpty(fundDistributions)) {
+      if (subtotal == 0d) {
+        validateZeroPrice(fundDistributions);
+        return;
       }
+      BigDecimal remainingPercent = ONE_HUNDRED_PERCENT;
+      for (FundDistribution fundDistribution : fundDistributions) {
+        FundDistribution.DistributionType dType = fundDistribution.getDistributionType();
+        if (dType == PERCENTAGE) {
+          remainingPercent = remainingPercent.subtract(BigDecimal.valueOf(fundDistribution.getValue()));
+        } else {
+          Double value = fundDistribution.getValue();
+          BigDecimal percentageValue = BigDecimal.valueOf(value)
+            .divide(BigDecimal.valueOf(subtotal), 15, HALF_EVEN)
+            .movePointRight(2);
+          remainingPercent = remainingPercent.subtract(percentageValue);
+        }
+      }
+      checkRemainingPercentMatchesToZero(remainingPercent, subtotal);
     }
   }
-
-  private boolean isFundDistributionSummaryNotValid(Adjustment adjustment, CurrencyUnit currencyUnit) {
-    return ObjectUtils.notEqual(sumMixedDistributions(adjustment.getFundDistributions(), adjustment.getValue(), currencyUnit), adjustment.getValue());
+  public static void validateZeroPrice(List<FundDistribution> fdList) {
+   FundDistribution.DistributionType firstFdType = fdList.get(0).getDistributionType();
```

Comment:
validate that the list is not empty before retrieving the first item.

Reply:
we are validating list in caller method & i have modified this method as private now.